Cheetah/legacy_parser.py

```diff
@@ -137,24 +137,22 @@ def _unescapeDirectives(s):
     'implements': 'eatImplements',
     'slurp': 'eatSlurp',
     'py': None,
-    'call': 'eatCall',
     'attr': 'eatAttr',
     'block': 'eatBlock',
     'end': 'eatEndDirective',
 }
 
-endDirectiveNamesAndHandlers = {
-    'def': 'handleEndDef',
-    'compiler-settings': None,
-    'block': None,
-    'call': None,
-    'while': None,
-    'for': None,
-    'if': None,
-    'try': None,
-    'filter': None,
-    'with': None,
-}
+CLOSABLE_DIRECTIVES = frozenset({
+    'block', 'compiler-settings', 'def', 'for', 'if', 'try', 'while', 'with',
```

Comment:
and `#def`, idk why `#compiler-settings` is indenting it really shouldn't be :)

That list might actually be somewhat bogus, I'm going to try taking some things out and seeing what fails